src/qmapshack/poi/CPoiCategory.cpp

```diff
@@ -19,17 +19,17 @@
 #include "CPoiCategory.h"
 #include "poi/CPoiPropSetup.h"
 
-CPoiCategory::CPoiCategory(const QString& categoryName, const QString& displayName, Qt::CheckState checkedState, CPoiCategory *parent)
-    : QTreeWidgetItem(parent), categoryName(categoryName)
+CPoiCategory::CPoiCategory(const QString& categoryName, uint categoryID, Qt::CheckState checkedState, CPoiCategory *parent)
+    : QTreeWidgetItem(parent), categoryName(categoryName), categoryID(categoryID)
 {
-    setData(CPoiPropSetup::eTreeColumnDisplayName, Qt::DisplayRole, displayName);
+    setData(CPoiPropSetup::eTreeColumnDisplayName, Qt::DisplayRole, categoryName);
```

Comment:
Why do you still use setData?

Reply:
Didn't you tell me in one of the last PRs (can't find it right now) to create an own class for the POI Categories and 'hide' the `setData` in there?

Reply:
I told you to get rid of setData(), by overriding QTreeWidgetItem and using member variables.

Reply:
Like this with `setText`?

Reply:
You already store `categoryName` as private member. There is no need to store it another time in that cryptic data storage of `QTreeWidgetItem`

Reply:
Yes, but I somehow need to set the text that is displayed to the user, don't I? That's why I now use `setText` instead of `setData`.

Reply:
:facepalm: ok, I should look into the code not just the commented code. All good..